sources/web/datalab/polymer/components/datalab-data/datalab-data.ts

```diff
@@ -0,0 +1,210 @@
+    this._collectionsList = [];
+    const collectionsElement = this.shadowRoot.querySelector('#collections')
+    if (collectionsElement) {
+      collectionsElement.addEventListener('itemDoubleClick',
+                                    this._collectionsDoubleClicked.bind(this));
+      collectionsElement.addEventListener('selected-indices-changed',
+                                    this._collectionsSelectionChanged.bind(this));
+    }
+
+    this._datasetsList = [];
+    const datasetsElement = this.shadowRoot.querySelector('#datasets')
+    if (datasetsElement) {
+      datasetsElement.addEventListener('itemDoubleClick',
+                                    this._datasetsDoubleClicked.bind(this));
+      datasetsElement.addEventListener('selected-indices-changed',
+                                    this._datasetsSelectionChanged.bind(this));
+    }
+  }
+
+  _generateFakeCollectionsListForTesting() {
+    const collectionsList = [];
+    const count = Math.floor(Math.random() * 20);
+    for (let i=0; i < count; i++) {
+      collectionsList.push(this._generateFakeCollectionForTesting());
+    }
+    return collectionsList;
+  }
+
+  _generateFakeCollectionForTesting() {
+    const fakeCollection = {
+      name: 'fakeCollection' + Math.floor(Math.random() * 1000000),
+      status: 'fake',
+    };
+    return fakeCollection;
+  }
+
+  _generateFakeDatasetsListForTesting() {
+    const datasetsList = [];
+    const count = Math.floor(Math.random() * 20);
+    for (let i=0; i < count; i++) {
+      datasetsList.push(this._generateFakeDatasetForTesting());
+    }
+    return datasetsList;
+  }
+
+  _generateFakeDatasetForTesting() {
+    const fakeDataset = {
+      name: 'fakeDataset' + Math.floor(Math.random() * 1000000),
+      status: 'fake',
+    };
+    return fakeDataset;
+  }
+
+  /** Reads the user's query values, queries for datasets, and updates the results. */
+  _search() {
+    this._collectionsList = this._generateFakeCollectionsListForTesting();
+    this._renderCollectionsList();
+  }
+
+  /**
+   * Creates a new ItemListRow object for each entry in the collections list, and sends
+   * the created list to the item-list to render.
+   */
+  _renderCollectionsList() {
+    this.$.collections.rows = this._collectionsList.map(collection => {
+      return {
+        firstCol: collection.name,
+        secondCol: collection.status,
+        icon: 'folder',
+        selected: false
+      };
+    });
+  }
+
+  _showDatasetsForCollection(collection: Collection) {
+    console.log('== collection selected:', collection);
+    if (!collection.datasets) {
+      collection.datasets = this._generateFakeDatasetsListForTesting();
+    }
+    this._datasetsList = collection.datasets;
+    this._renderDatasetsList();
+  }
+
+  _clearDatasetsList() {
+    this._datasetsList = [];
+    this._renderDatasetsList();
+  }
+
+  _showDetailsForDataset(dataset: Dataset) {
+    this.$.detailsPane.innerHTML = 'Selected dataset: ' + dataset.name;
```

Comment:
I expect I will be adding some kind of table-details element. I'll figure that out later.